ReactQt/runtime/src/componentmanagers/scrollviewmanager.cpp

```diff
@@ -153,8 +219,14 @@ void ScrollViewManager::configureView(QQuickItem* view) const {
     connect(view, SIGNAL(flickEnded()), SLOT(momentumScrollEnd()));
 }
 
-QString ScrollViewManager::qmlComponentFile() const {
-    return "qrc:/qml/ReactScrollView.qml";
+QString ScrollViewManager::qmlComponentFile(const QVariantMap& properties) const {
+
+    return properties.value("enableArrayScrollingOptimization", false).toBool() ? "qrc:/qml/ReactScrollListView.qml"
```

Comment:
@MaxRis, do we need to support both type of ScrollViews? Maybe we can go with new one in all cases?

Reply:
@vkjr that's a drawback of current optimization. It will work if array of children has ScrollView as a direct parent:
<ScrollView> <ArrayOfScrollableChildren /> </ScrollView>

and will not work in any other case, for example:
<ScrollView> <View><ArrayOfScrollableChildren /></View> </ScrollView>